src/AspNetCoreModuleV2/AspNetCore/src/proxymodule.cxx

```diff
@@ -126,13 +109,10 @@ ASPNET_CORE_PROXY_MODULE::OnExecuteRequestHandler(
             goto Finished;
         }
 
+        IREQUEST_HANDLER* pHandler;
         // Create RequestHandler and process the request
-        hr = pApplication->CreateHandler(pHttpContext, &m_pHandler);
-
-        if (FAILED(hr))
-        {
-            goto Finished;
-        }
+        FINISHED_IF_FAILED(pApplication->CreateHandler(pHttpContext, &pHandler));
+        m_pHandler.reset(pHandler);
```

Comment:
Why use reset here?

Reply:
You can't just assign unique_ptr it's either calling ctor or calling `.reset` and reset is much shorter.